### ISSUE TITLE
Flutter UI is extremely slow

### ISSUE THREAD

**Issue by kirgene** [flutter/flutter#26688]
I've installed [Flutter Gallery](https://play.google.com/store/apps/details?id=io.flutter.demo.gallery) app on my Android phone and noticed that animations, scrolling and transitions are lagging.

Interesting observation: often once particular animation was performed it not longer lags. Same applies to scrolling: once I scroll to the end of the list scrolling starts working fine.

Seems that there is some kind of cache that is populated and only after that acceptable performance is achieved.

Here is a demo:
https://drive.google.com/open?id=1ZXKLWQEYLpOcjd5h7ndrMqHhC1GW37Rg
Notice how slow is tap animation when I press the button first time.
Also the last screen (Contact Profile) is unusable: scrolling and overall performance is extremely slow.

I've also tested this app on Android Tablet and got the same behavior.

**Comment by zoechi** [comment]
What devices were the Android phone and Android tablet exactly?

**Comment by zoechi** [comment]
Looks like #813

**Comment by kirgene** [comment]
I've added phone and tablet info into issue description.

**Comment by zoechi** [comment]
@kirgene thanks
You saw this comment https://github.com/flutter/flutter/issues/813#issuecomment-454880245 in the issue linked to above?

**Comment by kirgene** [comment]
Yes, I saw it. Hope that fixing #813 also fix all related issues.

**Comment by luojieerr** [comment]
On some outdated devices, this problem still exists

**Comment by ivan-avalos** [comment]
I'm experiencing the same issue with Samsung Galaxy S4 with Android 4.4.4. I was testing with other apps and I found them really, really, really, really slow. Flutter Gallery didn't even open! It always crashes after 2 seconds of being opened. Flutter will never beat native performance.

**Comment by DeToxer** [comment]
Have you tried running the app in release mode? Debug mode vs release mode is night and day when it comes to performance.

**Comment by JamesHealdUK** [comment]
I can vouch for the same issues on a OnePlus 3 running Nougat 7.01

**Comment by revrkk** [comment]
Yes! I tried the same app in release mode. It is not laggy now.

(Release Mode -> `$ flutter run --release`)

**Comment by ivan-avalos** [comment]
> Yes @DeToxer ! I tried the same app in release mode. It is not laggy now.
> 
> (Release Mode -> `$ flutter run --release`)

I'm running Flutter Gallery and some Flutter apps in release mode in my iPhone 6S Plus running iOS 13.2 and yet, UI performance is awful, animations are laggy AF.

**Comment by fabien-ml** [comment]
Same here, running Flutter Gallery downloaded from the Play Store on a Galaxy A3 (2017) and some animation are laggy.

**Comment by adeolaex** [comment]
So like..... any news ?

**Comment by liyuqian** [comment]
Seems like a duplicate of https://github.com/flutter/flutter/issues/32170 and we've just provided a solution for Android through the closing of https://github.com/flutter/flutter/issues/53607. See the documentation on https://github.com/flutter/flutter/wiki/Reduce-shader-compilation-jank-using-SkSL-warm-up

**Comment by liyuqian** [comment]
CC @willlarche @clocksmith @guidezpl to see if the Gallery app could use the SkSL warm-up in newer releases, and set up proper CI tests to track the performance.

**Comment by Rayanfer32** [comment]
Flutter gallery UI animations are a lot janky .... Im on Android 10 and there id HWUI profiling for native apps, which shows processing loads bar graph even for react native apps....But crazy thing is Flutter apps doesnt show HWUI graph updates .
![Screenshot_20201207-002724](https://user-images.githubusercontent.com/37145078/101289671-f62bab80-3823-11eb-867d-162dd0bd6e2b.png)

**Comment by guidezpl** [comment]
@Rayanfer32 Hello, can you please provide debugging information (which device you're running on, which mode you're using (debug/profile, play store version, etc.)?

**Comment by Rayanfer32** [comment]
> @Rayanfer32 Hello, can you please provide debugging information (which device you're running on, which mode you're using (debug/profile, play store version, etc.)?

![photo_2020-12-07_16-58-10](https://user-images.githubusercontent.com/37145078/101345669-3eef5e80-383c-11eb-983a-99bfe9f16c0d.jpg)

**Comment by Rayanfer32** [comment]
The Lag is pretty much similar to people experiencing lags in Latest **Google Pay App** based on flutter.From my experience related to **HWUI not being used** for UI rendering

**Comment by lukaszciastko** [comment]
I just installed Flutter Gallery (in release mode) on my iPad Air 2020, and you can see it’s dropping frames every time I open a new page:

https://user-images.githubusercontent.com/964414/107818208-8e3f9700-6d77-11eb-8e2d-ff64d2b4a418.MP4

I get the same experience regardless of what I open in the app. It’s the same for sample apps.

**Comment by willlarche** [comment]
Refresh my memory. Did we ever implment shader warmup in here? @clocksmith

**Comment by lukaszciastko** [comment]
One more, from my iPhone 12 Pro Max:

https://user-images.githubusercontent.com/964414/107819562-aadcce80-6d79-11eb-954f-cade201408a4.MOV

It gets better second time the animation runs, but the first impresion is pretty bad...

**Comment by guidezpl** [comment]
@willlarche no shader warmup, the process wasn't scalable enough (yet) to handle all different types of devices and be created automatically

**Comment by willlarche** [comment]
Sounds like we could accept a community contrib on it if anyone wants to do
it manually.

**Comment by peterpan21** [comment]
Please flutter team, look into this and fix it soon. People all over the Internet have similar problems and complain about not being able to release their products.

**Comment by jh97uk** [comment]
> Please flutter team, look into this and give it a high priority. People all over the Internet have similar problems and complain about not being able to release their products. This and similar issues seriously diminish the quality of Flutter as a whole. People even state that they would not choose Flutter again for another project solely because of this issue.

This is personally why I'll likely never be using a platform like fluttar to develop my apps. Its seriously frustrating. The promise is boundless, especially for the kinds of projects I make. But the performance issue is time and time again the thing that always has me returning to vanilla native.

Its weird though, because I talk to people who say that their experience with performance with things like Flutter or React Native are silky smooth, but whenever I try example apps or apps developed with them, the perormance is always sub par. 

I think people majorly underestimate the importance of a silky smooth UI. No, 30 FPS when an openning animation is played is _not_ enough. There's a huge difference when you compare them one after the other.

I can only dream that one day the formula will be cracked for decent performance with cross platform dev :(

**Comment by clocksmith** [comment]
https://github.com/flutter/gallery/issues/354 has more info on adding this by way of an automated performance test
